src/btree/bt_delete.c

```diff
@@ -402,23 +397,24 @@ __wt_delete_page_instantiate(WT_SESSION_IMPL *session, WT_REF *ref)
                     upd->next = ins->upd;
                     ins->upd = upd;
 
-                    if (page_del != NULL)
-                        page_del->update_list[count++] = upd;
+                    if (update_list != NULL)
+                        update_list[count++] = upd;
                 }
         }
     }
-
     __wt_cache_page_inmem_incr(session, page, total_size);
 
+    /*
+     * We no longer need the WT_PAGE_DELETED structure, all of its information should have been
+     * transferred to the list of WT_UPDATE structures (if any).
+     */
+    __wt_overwrite_and_free(session, ref->ft_info.del);
```

Comment:
Should this be a `__wt_free`?

Reply:
Those are not `__wt_free()` because I want to make sure that we drop core if we access them in the "wrong way", that is, after the `WT_REF.state` has changed to an in-memory state.